src/cmap/auth/mongodb_oidc.ts

```diff
@@ -83,20 +93,25 @@ export interface Workflow {
 
 /** @internal */
 export const OIDC_WORKFLOWS: Map<ProviderName, Workflow> = new Map();
-OIDC_WORKFLOWS.set('callback', new CallbackWorkflow());
-OIDC_WORKFLOWS.set('aws', new AwsServiceWorkflow());
-OIDC_WORKFLOWS.set('azure', new AzureServiceWorkflow());
+OIDC_WORKFLOWS.set('automated_callback', new AutomatedCallbackWorkflow());
+OIDC_WORKFLOWS.set('human_callback', new HumanCallbackWorkflow());
+OIDC_WORKFLOWS.set('test', new TokenMachineWorkflow());
+OIDC_WORKFLOWS.set('azure', new AzureMachineWorkflow());
+OIDC_WORKFLOWS.set('gcp', new GCPMachineWorkflow());
 
 /**
  * OIDC auth provider.
  * @experimental
  */
 export class MongoDBOIDC extends AuthProvider {
+  cache?: TokenCache;
```

Comment:
Are there any scenarios where `cache` will not be provided?  If there aren't, it might be nicer to make it required.  This propagates through the workflows too, because if the cache is always present on the provider, it will always be provided to workflow.execute() or workflow.reauthenticate().

Reply:
This was also refactored to always be present and is now a property of the workflows themselves, and no longer needs to be passed though.